src/vaadin-grid-dynamic-columns-mixin.js

```diff
@@ -88,10 +88,14 @@ export const DynamicColumnsMixin = (superClass) =>
           this._rowDetailsTemplate = rowDetailsTemplate;
         }
 
-        if (
-          info.addedNodes.filter(this._isColumnElement).length > 0 ||
-          info.removedNodes.filter(this._isColumnElement).length > 0
-        ) {
+        const removedColumns = info.removedNodes.filter(this._isColumnElement);
+        if (info.addedNodes.filter(this._isColumnElement).length > 0 || removedColumns.length > 0) {
+          this.__removeSorters(
+            removedColumns.reduce((sorters, column) => sorters.concat(column.__getColumnSorters()), [])
+          );
+          this.__removeFilters(
+            removedColumns.reduce((filters, column) => filters.concat(column.__getColumnFilters()), [])
+          );
```

Comment:
The removed nodes contain the removed sorters/filters (as child elements). You shouldn't need to add any logic to vaadin-grid-column since you can get the removed sorters here instead.

Querying by tag name is also fragile (I remember a user who defined their own tag names for sorters). Shouldn't you be able to access all removed sorters here with `this._sorters.filter((n) => n.isConnected)`?

Reply:
I refactored not to query tag names, but `this._sorters.filter((n) => !n.isConnected)` was not enough.